api4/resolver_channel_member_test.go

```diff
@@ -0,0 +1,355 @@
+	th.App.AddUserToChannel(th.BasicUser, ch2, false)
+
+	// Creating some msgcount
+	th.CreateMessagePostWithClient(th.Client, th.BasicChannel, "basic post")
+	th.CreateMessagePostWithClient(th.Client, ch1, "ch1 post")
+
+	var q struct {
+		ChannelMembers []struct {
+			Channel struct {
+				ID          string            `json:"id"`
+				CreateAt    float64           `json:"createAt"`
+				UpdateAt    float64           `json:"updateAt"`
+				Type        model.ChannelType `json:"type"`
+				DisplayName string            `json:"displayName"`
+				Name        string            `json:"name"`
+				Header      string            `json:"header"`
+				Purpose     string            `json:"purpose"`
+				Team        struct {
+					ID string `json:"id"`
+				} `json:"team"`
+			} `json:"channel"`
+			User struct {
+				ID        string `json:"id"`
+				Username  string `json:"username"`
+				Email     string `json:"email"`
+				FirstName string `json:"firstName"`
+				LastName  string `json:"lastName"`
+				NickName  string `json:"nickname"`
+			} `json:"user"`
+			Roles []struct {
+				ID            string   `json:"id"`
+				Name          string   `json:"Name"`
+				Permissions   []string `json:"permissions"`
+				SchemeManaged bool     `json:"schemeManaged"`
+				BuiltIn       bool     `json:"builtIn"`
+			} `json:"roles"`
+			LastViewedAt     float64         `json:"lastViewedAt"`
+			LastUpdateAt     float64         `json:"lastUpdateAt"`
+			MsgCount         float64         `json:"msgCount"`
+			MentionCount     float64         `json:"mentionCount"`
+			MentionCountRoot float64         `json:"mentionCountRoot"`
+			NotifyProps      model.StringMap `json:"notifyProps"`
+			SchemeGuest      bool            `json:"schemeGuest"`
+			SchemeUser       bool            `json:"schemeUser"`
+			SchemeAdmin      bool            `json:"schemeAdmin"`
+			Cursor           string          `json:"cursor"`
+		} `json:"channelMembers"`
+	}
+
+	t.Run("all", func(t *testing.T) {
+		input := graphQLInput{
+			OperationName: "channelMembers",
+			Query: `
+	query channelMembers {
+	  channelMembers(userId: "me") {
+	  	channel {
+		  	id
+		  	createAt
+		  	updateAt
+		  	type
+		    displayName
+		    name
+		    header
+		    team {
+		    	id
+		    }
+	  	}
+	  	user {
+	  		id
+	  		username
+	  		email
+	  	}
+	  	msgCount
+	  	mentionCount
+	  	mentionCountRoot
+	  	schemeGuest
+	  	schemeUser
+	  	schemeAdmin
+	  	cursor
+	  }
+	}
+	`,
+		}
+
+		resp, err := th.MakeGraphQLRequest(&input)
+		require.NoError(t, err)
+		require.Len(t, resp.Errors, 0)
+		require.NoError(t, json.Unmarshal(resp.Data, &q))
+		assert.Len(t, q.ChannelMembers, 9)
+
+		numPrivate := 0
+		numPublic := 0
+		numOffTopic := 0
+		numTownSquare := 0
+		for _, ch := range q.ChannelMembers {
+			assert.NotEmpty(t, ch.Channel.ID)
+			assert.NotEmpty(t, ch.Channel.Name)
+			assert.NotEmpty(t, ch.Channel.CreateAt)
+			assert.NotEmpty(t, ch.Channel.UpdateAt)
```

Comment:
Wondering if we could leverage `require.NotZero` here to avoid many assertions. Eg:
```suggestion
			require.NotZero(t, ch.Channel)
```

Reply:
That would be slightly incorrect because in GraphQL, we are specifically fetching a set of fields. So that's why the test needs to check _exactly_ whether those fields are set or not. The `NotZero`  check will only verify whether it's a zero value or not, but won't check exactly which fields are set.